libazureinit/src/provision/ssh.rs

```diff
@@ -43,123 +45,225 @@ pub(crate) fn provision_ssh(
     Ok(())
 }
 
-pub(crate) fn get_authorized_keys_path_from_sshd() -> io::Result<Option<PathBuf>>
-{
-    let sshd_output = Command::new("sshd").arg("-T").output()?;
-    if !sshd_output.status.success() {
-        return Err(io::Error::new(
-            io::ErrorKind::Other,
-            format!("sshd -T failed with status: {}", sshd_output.status),
-        ));
+fn get_sshd_g_output() -> io::Result<Output> {
+    Command::new("sshd").arg("-G").output()
+}
+
+fn get_authorized_keys_path_from_sshd(
+    sshd_config_command_runner: impl Fn() -> io::Result<Output>,
+) -> Option<String> {
+    let output: Output = match run_sshd_command(sshd_config_command_runner) {
+        Some(output) => output,
+        None => {
+            info!("Unable to determine authorizedkeysfile from sshd -G");
```

Comment:
Do we need this log line? The called function includes logging for all branch arms that return `None` at error level so it feels a bit repetitive (although neither line indicates it failed to determine authorizedkeyfile).

If you opt to drop it, you can also replace this match block with
```Rust
let output = run_sshd_command(...)?;
```

Reply:
done, thanks!